code/modules/mob/living/simple_animal/hostile/space_dragon.dm

```diff
@@ -360,31 +375,39 @@
  * Empowers and depowers Space Dragon after a successful rift charge.
  * Empowered, Space Dragon regains all his health and becomes temporarily faster for 30 seconds, along with being tinted red.
  * Depowered simply resets him back to his default state.
+ * Arguments:
+ * * is_empowered - Whether or not we're adding the buff or removing the buff
+ * * is_permanent - Only applies if is_empowered is true, whether or not the buff should be removed after a time.
  */
-/mob/living/simple_animal/hostile/space_dragon/proc/rift_empower(is_empowered)
+/mob/living/simple_animal/hostile/space_dragon/proc/rift_empower(is_empowered, is_permanent)
```

Comment:
Can you make this a bitfield instead?

[I don't particularly like boolean parameters anyway](https://medium.com/@amlcurran/clean-code-the-curse-of-a-boolean-parameter-c237a830b7a3), but if we're going to have them they should at least be readable.

Alternatively, splitting up the bools into separate procs is more prefered if you're willing to put in more work, especially `is_empowered` which looks to alter the *entire* behavior.